src/dataflow/server.rs

```diff
@@ -677,18 +681,24 @@ where
             } => {
                 let mut timestamps = self.ts_histories.borrow_mut();
                 if let Some(entries) = timestamps.get_mut(&id) {
-                    let ts = match entries.get_mut(&pid) {
-                        Some(ts) => ts,
-                        None => {
-                            entries.insert(pid.clone(), vec![]);
-                            entries.get_mut(&pid).unwrap()
-                        }
-                    };
-                    let last_offset = if let Some(offs) = ts.last() {
-                        offs.2
+                    let ts = entries.entry(pid).or_insert_with(Vec::new);
+                    let (last_timestamp, last_offset) = if let Some(offs) = ts.last() {
+                        (offs.1, offs.2)
                     } else {
-                        0
+                        (0, 0)
                     };
+                    assert!(
```

Comment:
Worth adding a comment here that the reason why this holds is become timestamp.rs always assigns a timestamp > 0?

Reply:
:+1: